pkg/crc/image/image.go

```diff
@@ -0,0 +1,112 @@
+package image
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"runtime"
+	"strings"
+
+	crcpreset "github.com/code-ready/crc/pkg/crc/preset"
+	"github.com/containers/image/v5/copy"
+	"github.com/containers/image/v5/signature"
+	"github.com/containers/image/v5/transports/alltransports"
+	"github.com/containers/image/v5/types"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+var (
+	registryURI = "quay.io"
+	orgName     = "crcont"
+)
+
+type OCIImageHandler struct {
+	ImageName string
+	ImageTag  string
+	Progress  io.Writer
+}
+
+func (img *OCIImageHandler) uri() string {
+	return strings.Join([]string{registryURI, orgName, img.ImageName}, "/")
+}
+
+func (img *OCIImageHandler) parse(image string) (types.ImageReference, error) {
+	return alltransports.ParseImageName(image)
+}
+
+func (img *OCIImageHandler) policyContext() (*signature.PolicyContext, error) {
+	policy := &signature.Policy{Default: []signature.PolicyRequirement{signature.NewPRInsecureAcceptAnything()}}
+	policyContext, err := signature.NewPolicyContext(policy)
+	if err != nil {
+		return nil, fmt.Errorf("error creating security context: %s", err.Error())
+	}
+
+	return policyContext, nil
+}
+
+func (img *OCIImageHandler) getSystemContext() *types.SystemContext {
+	return &types.SystemContext{
+		OSChoice: "linux",
+	}
+}
+
+// CopyImage pull the image from the registry and put it to local storage
+func (img *OCIImageHandler) CopyImage(destPath string) ([]byte, error) {
+	// Source Image from docker transport
+	srcImg := fmt.Sprintf("docker://%s:%s", img.uri(), img.ImageTag)
+	srcRef, err := img.parse(srcImg)
+	if err != nil {
+		return nil, fmt.Errorf("invalid source image name %s: %v", srcImg, err)
+	}
+
+	destImg := fmt.Sprintf("dir:%s", destPath)
+	destRef, err := img.parse(destImg)
+	if err != nil {
+		return nil, fmt.Errorf("invalid destination name %s: %v", destImg, err)
+	}
+
+	policyContext, err := img.policyContext()
+	if err != nil {
+		return nil, err
+	}
+
+	return copy.Image(context.Background(), policyContext,
+		destRef, srcRef, &copy.Options{
+			ReportWriter: img.Progress,
+			SourceCtx:    img.getSystemContext(),
+		})
```

Comment:
I would pass `Progress` as an argument to this method rather than having it in `OCIImageHandler` where it feels out of place. Also, `Progress` is slightly misleading, `ReportWriter` is more accurate.

Reply:
I don't see why if it feel out of place, we create imagehandler and run this method. I will change the Progress to `ReportWriter` if that make it more accurate. I used progress because it show the progress of image layer download/pull.

Reply:
> Progress is only useful to the Copy method, so it's not clear why we need to store it in the OCIImage type which can be used over other methods? Isn't it clearer to pass it to the only method which needs it?

@cfergeau make sense, will update this.